src/screens/mints/MintSheet.tsx

```diff
@@ -0,0 +1,861 @@
+  const [insufficientEth, setInsufficientEth] = useState(false);
+  const [showNativePrice, setShowNativePrice] = useState(false);
+  const [gasError, setGasError] = useState(false);
+  const currentNetwork =
+    RainbowNetworks.find(({ id }) => id === mintCollection.chainId)?.value ||
+    Network.mainnet;
+  const [ensName, setENSName] = useState<string>('');
+  const [mintStatus, setMintStatus] = useState<
+    'none' | 'minting' | 'minted' | 'error'
+  >('none');
+  const txRef = useRef<string>();
+
+  const { data: ensAvatar } = useENSAvatar(ensName, {
+    enabled: Boolean(ensName),
+  });
+
+  const [quantity, setQuantity] = useReducer(
+    (quantity: number, increment: number) => {
+      if (quantity === 1 && increment === -1) {
+        return quantity;
+      }
+      if (
+        maxMintsPerWallet &&
+        quantity === maxMintsPerWallet &&
+        increment === 1
+      ) {
+        return quantity;
+      }
+      return quantity + increment;
+    },
+    1
+  );
+
+  // if there is no max mint info, we fallback to 1 to be safe
+  const maxMintsPerWallet = Number(
+    mintCollection.publicMintInfo?.maxMintsPerWallet
+  );
+
+  const price = convertRawAmountToBalance(
+    mintCollection.publicMintInfo?.price?.amount?.raw || '0',
+    {
+      decimals: mintCollection.publicMintInfo?.price?.currency?.decimals || 18,
+      symbol: mintCollection.publicMintInfo?.price?.currency?.symbol || 'ETH',
+    }
+  );
+
+  // case where mint isnt eth? prob not with our current entrypoints
+  const mintPriceAmount = multiply(price.amount, quantity);
+  const mintPriceDisplay = convertAmountToBalanceDisplay(
+    multiply(price.amount, quantity),
+    {
+      decimals: mintCollection.publicMintInfo?.price?.currency?.decimals || 18,
+      symbol: mintCollection.publicMintInfo?.price?.currency?.symbol || 'ETH',
+    }
+  );
+
+  const priceOfEth = ethereumUtils.getEthPriceUnit() as number;
+
+  const nativeMintPriceDisplay = convertAmountToNativeDisplay(
+    parseFloat(multiply(price.amount, quantity)) * priceOfEth,
+    nativeCurrency
+  );
+
+  const {
+    updateTxFee,
+    startPollingGasFees,
+    stopPollingGasFees,
+    isSufficientGas,
+    isValidGas,
+    getTotalGasPrice,
+  } = useGas();
+
+  const imageUrl = maybeSignUri(mintCollection.image || '');
+  const { result: aspectRatio } = usePersistentAspectRatio(imageUrl || '');
+
+  // isMintingPublicSale handles if theres a time based mint, otherwise if there is a price we should be able to mint
+  const isMintingAvailable =
+    !(isReadOnlyWallet || isHardwareWallet) &&
+    (mintCollection.isMintingPublicSale || price) &&
+    !gasError;
+
+  const imageColor =
+    usePersistentDominantColorFromImage(imageUrl) ?? colors.paleBlue;
+
+  const sheetRef = useRef();
+  const yPosition = useSharedValue(0);
+
+  useFocusEffect(() => {
+    if (mintCollection.name && mintCollection.id) {
+      analyticsV2.track(event.nftMintsOpenedSheet, {
```

Comment:
i think in general we should decide if we want to trigger events before or after navigation occurs. like do we trigger it when the button is pressed that navs to the sheet, or when the sheet is rendered?

Reply:
just a random thought